### PR TITLE
Add Stape logging format.

### LINKED ISSUE #5: Stape Logging standart
It will be great if tag will support Stape Logging Standard.
Currently it supported by majority of sGTM tags.

https://github.com/stape-io/gtm-standards/blob/main/standards/GTMS-3.md

If it's good for you I can prepare PR.

### PR DESCRIPTION
**Pull request by Bukashk0zzz**
Add Stape logging format. 

Closes #5

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -587,33 +615,53 @@ ___SANDBOXED_JS_FOR_SERVER___
...
 const makeString = require('makeString');
 const Object = require('Object');
 const sendHttpRequest = require('sendHttpRequest');
-const setResponseBody = require('setResponseBody');
 const setResponseHeader = require('setResponseHeader');
 const setResponseStatus = require('setResponseStatus');
 const sha256Sync = require('sha256Sync');
 const testRegex = require('testRegex');
+const getRequestHeader = require('getRequestHeader');
+const getContainerVersion = require('getContainerVersion');
```
Comment (sahava) [inline template.tpl:630]: Move these two `const` declarations to correct positions in the API list (alphabetical order).

File: template.tpl [hunk 2]
```diff
@@ -587,33 +615,53 @@ ___SANDBOXED_JS_FOR_SERVER___
...
 
 const ENDPOINT = 'https://' + data.instanceName + '.piwik.pro/ppms.php';
 const LIBRARY_NAME = 'sgtm';
-const LIBRARY_VERSION = '1.0.0';
-const LOG_PREFIX = '[ppms_tag] ';
+const LIBRARY_VERSION = '1.0.1';
+
+const isLoggingEnabled = determinateIsLoggingEnabled();
```
Comment (sahava) [inline template.tpl:636]: Move up to list of consts, change to capital case, use renamed function:

```
const LOGGING_ENABLED = isLoggingEnabled();
```

File: template.tpl [hunk 3]
```diff
@@ -587,33 +615,53 @@ ___SANDBOXED_JS_FOR_SERVER___
...
+const traceId = isLoggingEnabled ? getRequestHeader('trace-id') : undefined;
```
Comment (sahava) [inline template.tpl:637]: Change to `const traceId = LOGGING_ENABLED ? ...`

File: template.tpl [hunk 4]
```diff
@@ -587,33 +615,53 @@ ___SANDBOXED_JS_FOR_SERVER___
...
+/**
+ * Returns if logging is enabled based on the logType value and the container version.
  *
- * @param {String} message - the message to be logged.
+ * @returns {boolean}
  */
-const log = msg => {
-  logToConsole(LOG_PREFIX + msg);
-};
+function determinateIsLoggingEnabled() {
```
Comment (sahava) [inline template.tpl:644]: Change to `const` like other function declarations and rename to `isLoggingEnabled`:

```
const isLoggingEnabled = () => {
```

File: template.tpl [hunk 5]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
+    .forEach(key => requestBody[key] = uiParamMap[key]);
 
 // Add the library name and version
 requestBody.ts_n = LIBRARY_NAME;
 requestBody.ts_v = LIBRARY_VERSION;
 
 requestBody = cleanObject(requestBody);
 
-log('Compiled request body: ' + JSON.stringify(requestBody));
+const headers = {
```
Comment (sahava) [inline template.tpl:807]: move this to where the other constants are declared, and use capital case 

```
const HEADERS = {
  ...
}
```

File: template.tpl [hunk 6]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
+  'content-type': 'application/x-www-form-urlencoded'
+};
+
+if (isLoggingEnabled) {
```
Comment (sahava) [inline template.tpl:811]: Change to `if (LOGGING_ENABLED) {`

File: template.tpl [hunk 7]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
-
-log('Encoded POST body: ' + postBody);
+    .slice(0, -1);
 
 sendHttpRequest(ENDPOINT, {
-  headers: {
-    'content-type': 'application/x-www-form-urlencoded'
-  }, 
-  method: 'POST', 
+  headers: headers,
```
Comment (sahava) [inline template.tpl:835]: Change to `headers: HEADERS`

File: template.tpl [hunk 8]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
+  method: 'POST',
   timeout: 1000
- }, postBody).then(response => {
+}, postBody).then(response => {
+  if (isLoggingEnabled) {
```
Comment (sahava) [inline template.tpl:839]: Change to `if (LOGGING_ENABLED) {`

**Review by sahava — CHANGES_REQUESTED** [review]

File: template.tpl [hunk 9]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
+const headers = {
+  'content-type': 'application/x-www-form-urlencoded'
+};
+
+if (isLoggingEnabled) {
+  logToConsole(
+      JSON.stringify({
+        Name: 'PiwikPro',
+        Type: 'Request',
+        TraceId: traceId,
```
Comment (sahava) [inline template.tpl:816]: Update to `TraceId: TRACE_ID`

File: template.tpl [hunk 10]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
+        EventName: requestBody.action_name || 'page_view',
+        RequestMethod: 'POST',
+        RequestUrl: ENDPOINT,
+        RequestHeaders: headers,
```
Comment (sahava) [inline template.tpl:820]: Update to `RequestHeaders: HEADERS`

File: template.tpl [hunk 11]
```diff
@@ -743,45 +791,72 @@ let requestBody = {};
...
+  method: 'POST',
   timeout: 1000
- }, postBody).then(response => {
+}, postBody).then(response => {
+  if (isLoggingEnabled) {
+    logToConsole(
+        JSON.stringify({
+          Name: 'PiwikPro',
+          Type: 'Response',
+          TraceId: traceId,
```
Comment (sahava) [inline template.tpl:844]: Update to `TraceId: TRACE_ID`

**Comment by Bukashk0zzz** [comment]
@sahava Thanks for the review and all your fixes.

**Review by abartczakpiwik — APPROVED** [review]

**Review by sahava — APPROVED** [review]

**Pull request merged**